packages/injective-cosmwasm/src/types.rs

```diff
@@ -142,17 +142,17 @@ impl<'de> Deserialize<'de> for SubaccountId {
         let subaccount_id = String::deserialize(deserializer)?;
 
         let is_nonce_derived = subaccount_id.len() <= MAX_SUBACCOUNT_NONE_LENGTH;
-        if is_nonce_derived && subaccount_id.parse::<u64>().is_err() {
+        if is_nonce_derived && (subaccount_id.is_empty() || subaccount_id.parse::<u64>().is_err()) {
             let error_message = "Invalid nonce: when subaccount_id is derived from nonce, it must be a valid u64";
             return Err(D::Error::custom(error_message));
         }
 
-        if !subaccount_id.starts_with("0x") {
+        if !is_nonce_derived && !subaccount_id.starts_with("0x") {
             let error_message = "Invalid prefix in deserialization: subaccount_id must start with 0x";
             return Err(D::Error::custom(error_message));
         }
 
-        if subaccount_id.len() != 66 {
+        if !is_nonce_derived && subaccount_id.len() != 66 {
             let error_message = "Invalid length in deserialization: subaccount_id must be exactly 66 characters";
             return Err(D::Error::custom(error_message));
         }
```

Comment:
I'd also add one more check in both places:
```
     if is_nonce_derived && subaccount_id.len() > 3  {
            let error_message = "Invalid length in deserialization: nonce-derived subaccount_id can have up to 3 digits";
            return Err(D::Error::custom(error_message));
        }
```

Reply:
that case is impossible, `is_nonce_derived` is only true if `subaccount_id.len() <= 3`